File: backend/src/main/java/com/youcode/MonSupplier/services/interfaces/AdminService.java
package com.youcode.MonSupplier.services.interfaces;


import com.youcode.MonSupplier.models.Enums.RoleUser;

import java.util.List;

public interface AdminService{
    List getAll(RoleUser roleUser);

    void deleteById(RoleUser roleUser, Long id);

    Object getById(RoleUser roleUser, Long id);

    Object create(RoleUser roleUser, Object object);

    Object update(RoleUser roleUser, Object object);
}

File: frontend 17/src/app/Services/Admin/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  private baseUrl = 'http://localhost:8080/api/Admin';

  constructor(private http: HttpClient) {}

  getAll(roleUser: string): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/all/${roleUser}`);
  }

  getById(roleUser: string, id: number): Observable<any> {
    return this.http.get<any>(`${this.baseUrl}/user/${roleUser}/${id}`);
  }

  deleteById(roleUser: string, id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/delete/${roleUser}/${id}`);
  }

  create(roleUser: string, object: any): Observable<any> {
    return this.http.post<any>(`${this.baseUrl}/create/${roleUser}`, object);
  }

  update(roleUser: string, object: any): Observable<any> {
    return this.http.put<any>(`${this.baseUrl}/update/${roleUser}`, object);
  }
}
